### ISSUE TITLE
Searching on index with spaces on its name crashes app

### ISSUE THREAD

**Issue by cruzdiego** [algolia/algoliasearch-client-swift#103]
There is an exception thrown when searching on an index with spaces on its name (e.g. "Index with space").

The solution is correctly implementing Helpers class's urlEncode() function:

Current implementation:
```
func urlEncode() -> String {
        let customAllowedSet = NSCharacterSet(charactersInString: "!*'();:@&=+$,/?%#[]").invertedSet
        return stringByAddingPercentEncodingWithAllowedCharacters(customAllowedSet)!
    }
```

Proposed solution:
```
func urlEncode() -> String {
        return stringByAddingPercentEncodingWithAllowedCharacters(.URLHostAllowedCharacterSet())!
    }
```

**Comment by clement-leprovost** [comment]
@cruzdiego Wow, nice catch! I spotted a few other inconsistencies, so I took the opportunity to refactor URL encoding. :)

A fix is ready, waiting for continuous integration (#104). Note: We have a few pending PRs, so the release may take a few days to follow.

**Comment by clement-leprovost** [comment]
Fixed in [version 3.4](https://github.com/algolia/algoliasearch-client-swift/releases/tag/3.4).
